### PR TITLE
Add VSCode native debug launcher that attaches to java process.

### PR DESCRIPTION
**Pull request by brettle**
Note that this includes PR #114.

Other notes:

In my experience on linux searching for "executor" brings up  the "Gradle Test Executor" java process of interest when debugging tests hanging in native code.

The sourceFileMap assumes that that wpilib source is installed under $HOME/git/allwpilib. If it isn't, you can still see stack traces, you just won't be able to see the wpilib source.

The "program" location probably needs a windows specific version, maybe with ${env:HOME} replaced with ${env:ALLUSERSPROFILE} or "C:\Users\Public"?

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -4,6 +4,28 @@
     // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
     "version": "0.2.0",
     "configurations": [
+        {
+            "name": "(gdb) Attach",
+            "type": "cppdbg",
+            "request": "attach",
+            "program": "${env:HOME}/wpilib/2024/jdk/bin/java",
```
Comment (CoolSpy3) [inline .vscode/launch.json:11]:
```suggestion
            "program": "${env:HOME}/wpilib/2024/jdk/bin/java",
            "windows": {
                "program": "${env:PUBLIC}\\wpilib\\2024\\jdk\\bin\\java"
            },
```

File: .vscode/launch.json [hunk 2]
```diff
@@ -4,6 +4,28 @@
...
+            "sourceFileMap": {
+                "/work/": "${env:HOME}/git/allwpilib/"
```
Comment (CoolSpy3) [inline .vscode/launch.json:13]: Personally, I'm not a fan of searching for the WPILib sources in a specific directory, as it's unlikely all users will store it in the same place. I'd like this to either be made into an input or tied to an environment variable.

**Review by CoolSpy3 — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 2 above)
Comment (brettle) [reply .vscode/launch.json]: Fixed.

**Review by CoolSpy3 — APPROVED** [review]
:+1:

**Pull request merged**